core/src/components/input/input.ionic.outline.scss

```diff
@@ -0,0 +1,162 @@
+ * The bottom content should never have
+ * a border with the outline style.
+ */
+:host(.input-fill-outline) .input-bottom {
+  border-top: none;
+}
+
+/**
+ * Outline inputs do not have a bottom border.
+ * Instead, they have a border that wraps the
+ * input + label.
+ */
+:host(.input-fill-outline) .input-wrapper {
+  border-bottom: none;
+}
+
+:host(.input-fill-outline.input-label-placement-stacked) .label-text-wrapper,
+:host(.input-fill-outline.input-label-placement-floating) .label-text-wrapper {
+  @include transform-origin(start, top);
+
+  position: absolute;
+
+  /**
+   * Label text should not extend
+   * beyond the bounds of the input.
+   */
+  max-width: calc(100% - var(--padding-start) - var(--padding-end));
+}
+
+/**
+ * The label should appear on top of an outline
+ * container that overlaps it so it is always clickable.
+ */
+:host(.input-fill-outline) .label-text-wrapper {
+  position: relative;
+}
+
+/**
+ * This ensures that the input does not
+ * overlap the floating label while still
+ * remaining visually centered.
+ */
+:host(.input-fill-outline.input-label-placement-stacked) input,
+:host(.input-fill-outline.input-label-placement-floating) input {
+  @include margin(6px, 0, 6px, 0);
+}
+
+
+// Input Fill: Outline, Outline Container
+// ----------------------------------------------------------------
+
+/**
+ * Note that while the ionic theme does not make use of the notch,
+ * thus avoiding the need for separate outline container parts at all,
+ * the element's structure has been kept the same to simplify the
+ * component render function and keep selectors consistent between
+ * the ionic and md themes.
+ */
+
+:host(.input-fill-outline) .input-outline-container {
+  @include position(0, 0, 0, 0);
+
+  display: flex;
+
+  position: absolute;
+
+  width: 100%;
+  height: 100%;
+}
+
+:host(.input-fill-outline) .input-outline-start,
+:host(.input-fill-outline) .input-outline-end {
+  pointer-events: none;
+}
+
+/**
+ * Each piece of the container should have a top and bottom
+ * border. This gives the appearance of a unified container
+ * with a border.
+ */
+:host(.input-fill-outline) .input-outline-start,
+:host(.input-fill-outline) .input-outline-notch,
+:host(.input-fill-outline) .input-outline-end {
+  border-top: var(--border-width) var(--border-style) var(--border-color);
+  border-bottom: var(--border-width) var(--border-style) var(--border-color);
+}
+
+/**
+ * Ensures long labels do not cause the notch to flow
+ * out of bounds.
+ */
+:host(.input-fill-outline) .input-outline-notch {
+  max-width: calc(100% - var(--padding-start) - var(--padding-end));
+}
+
+/**
+ * In the md theme, this element ensures that the notch
+ * uses the size of the scaled text so that the border
+ * cut-out is the correct width. However, a notch is not
+ * shown in the ionic theme, so we only need to make sure
+ * the text in this element is not visible or interactive.
+ */
+:host(.input-fill-outline) .notch-spacer {
+  opacity: 0;
```

Comment:
If the notch is not shown, why are not using `display: none` here?

Reply:
The notch was removed per other comments, so marking this as resolved.